src/fontra/views/editor/panel-transformation.js

```diff
@@ -531,63 +531,62 @@ export default class TransformationPanel extends Panel {
   }
 
   async doPathOperations(pathOperationFunc, undoLabel) {
+    const doUnion = pathOperationFunc === unionPath;
     let { point: pointIndices } = parseSelection(this.sceneController.selection);
     pointIndices = pointIndices || [];
 
     const positionedGlyph =
       this.sceneController.sceneModel.getSelectedPositionedGlyph();
 
-    const allContourIndices = [...range(positionedGlyph.glyph.path.numContours)];
-    const selectedContourIndices =
-      !pointIndices.length && undoLabel === "Remove overlaps"
-        ? allContourIndices
-        : [...getSelectionByContour(positionedGlyph.glyph.path, pointIndices).keys()];
-
-    if (!selectedContourIndices.length) {
-      return;
-    }
-
-    const notSelectedContourIndices = allContourIndices.filter(
-      (n) => !selectedContourIndices.includes(n)
+    const selectedContourIndicesMap = getSelectionByContour(
+      positionedGlyph.glyph.path,
+      pointIndices
     );
-    if (!notSelectedContourIndices.length && undoLabel !== "Remove overlaps") {
-      return;
-    }
+    const isContourSelected =
+      pointIndices.length || !doUnion
+        ? (i) => selectedContourIndicesMap.has(i)
+        : (i) => true;
 
     const editLayerGlyphs = this.sceneController.getEditingLayerFromGlyphLayers(
       positionedGlyph.varGlyph.glyph.layers
     );
 
+    const contoursToBeDeleted = []; // this is only needed for doUnion
     const layerPaths = await mapObjectValuesAsync(
       editLayerGlyphs,
       async (layerGlyph) => {
-        const path = layerGlyph.path;
+        const path = layerGlyph.path.copy();
         const selectedContoursPath = new VarPackedPath();
         const unselectedContoursPath = new VarPackedPath();
 
         for (const contourIndex of range(path.numContours)) {
-          if (selectedContourIndices.includes(contourIndex)) {
+          if (isContourSelected(contourIndex)) {
             selectedContoursPath.appendContour(path.getContour(contourIndex));
+            contoursToBeDeleted.push(contourIndex);
```

Comment:
You are doing this for each layer, so this likely breaks multi source editing. Also, it is literally the list of keys of `selectedContourIndicesMap`, so there is really no need to do this here.

Reply:
done, please see: https://github.com/googlefonts/fontra/pull/1582/commits/dc01cb28bae3b80e17cbf6243667aacca2e5f57f